### PR TITLE
Adjusts how tags are capitalized.

### LINKED ISSUE #2172: Reader: Adjust how topics are capitalized
Capitalizing a topic can result in a properly formatted topic like "WordPress" being displayed as "Wordpress" (capital P dangit!).  Tweak the logic so we only capitalize when necessary.

### PR DESCRIPTION
**Pull request by aerych**
Closes #2172

### CODE REVIEW

**Comment by aerych** [comment]
cc @nbradbury for a very violent review. :)

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java [hunk 1]
```diff
@@ -74,10 +75,17 @@ public String getCapitalizedTagName() {
         if (tagName == null) {
             return "";
         }
-        // HACK to allow iPhone, iPad, iEverything else
-        if (tagName.startsWith("iP")) {
+        // If already uppercase, assume correctly formatted
+        if (Character.isUpperCase(tagName.charAt(0))) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: Although it's unlikely, it is possible that tagName will be "" in which case `charAt(0)` will cause a null pointer exception.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java [hunk 2]
```diff
@@ -74,10 +75,17 @@ public String getCapitalizedTagName() {
...
             return tagName;
         }
+        // Accounts for iPhone, ePaper, etc.
+        if (tagName.length() > 1 &&
+                Character.isUpperCase(tagName.charAt(0)) &&
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: I think these are reversed. Should be checking if the first character is lowercase and the second is uppercase.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java (hunk 1 above)
Comment (aerych) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: What's funny is I checked for string length in the iOS version.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java (hunk 2 above)
Comment (aerych) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: Indeed.

**Comment by aerych** [comment]
@nbradbury take two?

**Comment by nbradbury** [comment]
Looks good @aerych - I'll merge it.

**Pull request merged**